tools/build_pak_files.py

```diff
@@ -0,0 +1,100 @@
+    unreal_pak_exe    = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Win64", "UnrealPak.exe"))
+    
+    assert os.path.exists(args.unreal_project_dir)
+    _, project = os.path.split(args.unreal_project_dir)
+    uproject   = os.path.realpath(os.path.join(args.unreal_project_dir, project + ".uproject"))
+    
+    for platform in args.platforms:
+
+        cmd = [ 
+            unreal_editor_exe,
+            uproject,
+            "-run=Cook",
+            "-TargetPlatform=" + platform + "NoEditor",
+            "-fileopenlog",
+            "-ddc=InstalledDerivedDataBackendGraph",
+            "-unversioned",
+            "-stdout",
+            "-FullStdOutLogOutput",
+            "-CrashForUAT",
+            "-unattended",
+            "-NoLogTimes",
+            "-UTF8Output"
+        ]
+        print(f"[SPEAR | build_pak_files.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd, stdout=sys.stdout.buffer, stderr=sys.stdout.buffer)
+        assert cmd_result.returncode == 0
+
+        content_dirs = [
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", project,  "Content")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "Animation")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "BasicShapes")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Texturing")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Math")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "Functions", "MaterialLayerFunctions")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Utility")),
+            os.path.realpath(os.path.join(args.unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor", "Engine", "Content", "EngineResources"))
+        ]
+
+        # construct output pak file path
+        output_pak = os.path.join(args.output_dir, f"Map_{args.scene_id}_{platform}.pak")
+
+        # temporary text file used to generate the final pak file
+        pak_txt_path = os.path.join(args.output_dir, f"pak_data_{platform}.txt")
+
+        if os.path.exists(pak_txt_path):
+            os.remove(pak_txt_path)
+
+        for dir in content_dirs:
+            with open(pak_txt_path, mode='a') as fout:
+                for fpath in glob.glob(dir + "\**\*.*", recursive=True):
```

Comment:
use os.path.join instead of this style of concatenating

Reply:
Done